apps/web-giddh/src/app/expenses/components/pending-list/pending-list.component.ts

```diff
@@ -284,4 +296,43 @@ export class PendingListComponent implements OnInit, OnChanges {
             }, 500);
         }
     }
+
+    /**
+     * Prepares the entry
+     *
+     * @private
+     * @param {PettyCashResonse} res Petty cash details
+     * @memberof PendingListComponent
+     */
+    private prepareEntryAgainstObject(res: ExpenseResults): void {
+        let cashOrBankEntry = res?.baseAccount ? this.isCashBankAccount(res.baseAccount) : false;
+        if (res?.entryType === 'sales') {
+            this.entryAgainstObject.base = cashOrBankEntry ? 'Receipt Mode' : 'Debtor Name';
+            this.entryAgainstObject.against = cashOrBankEntry ? 'Entry against Debtor' : 'Cash Sales';
+        } else if (res?.entryType === 'expense') {
+            this.entryAgainstObject.base = cashOrBankEntry ? 'Payment Mode' : 'Creditor Name';
+            this.entryAgainstObject.against = cashOrBankEntry ? 'Entry against Creditors' : 'Cash Expenses';
+        } else {
+            this.entryAgainstObject.base = 'Deposit To';
+            this.entryAgainstObject.against = null;
+        }
+
+        this.entryAgainstObject.model = res?.baseAccount?.uniqueName;
+        this.entryAgainstObject.name = res?.baseAccount?.name;
+    }
+
+    /**
+     * Returns true, if the account belongs to cash or bank account
+     *
+     * @private
+     * @param {any} particular Account unique name
+     * @returns {boolean} Promise to carry out further operations
```

Comment:
Do correct the JS doc the method doesn't return a promise but the doc states that.